src/codons.py

```diff
@@ -60,15 +75,27 @@ def translate_codons(codons: list[str]) -> list[str]:
     True
 
     """
-    # FIXME: Implement the function
-    return []
-
 
-def translate_dna(dna: str) -> str:
+    amino_list = []
+    if codons_list == None:
+        amino_list = None
+    if codons_list != None:
+        for codon in codons_list:
+            if codon.upper() not in CODON_MAP.keys():
```

Comment:
You should just test `if key in table:` not `if key in table.keys()`. Testing in a table is highly efficient, but if you get the keys out in a list it isn't any more.